### ISSUE TITLE
Disallow using findBy* inside a waitFor block

### ISSUE THREAD

**Issue by neriyarden** [testing-library/eslint-plugin-testing-library#910]
### Name for new rule

no-findby-in-waitfor

### Description of the new rule

This new rule should prevent the usage of a `findBy`* method inside a `waitFor`.

### Testing Library feature

`waitFor` and `findBy`

### Testing Library framework(s)

all

### What category of rule is this?

Suggests an alternate way of doing something

### Code examples

**Incorrect usage**
```ts
await waitFor(() => {
  const button = await findByRole("button", { name: "Submit" });
  expect(button).toBeInTheDocument()
};
```

**Correct usage**
```ts
  const button = await findByRole("button", { name: "Submit" });
  expect(button).toBeInTheDocument()
```

### Do you want to submit a pull request to make the new rule?

Yes

**Comment by Belco90** [comment]
Thanks for your idea! I'd update the current [`prefer-find-by`](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/prefer-find-by.md) to report this case, instead of creating a new rule. We would need to update the rule description so it indicates that suggests using `findBy` queries instead of `waitFor` + queries to wait for elements.
